PubnativeLite/PubnativeLite/Ad Request/HyBidAdRequest.m

```diff
@@ -433,6 +470,92 @@ - (void)processXmlResponseWithData:(NSData *)data {
     }
 }
 
+- (void)processVASTXmlResponseWithData:(NSData *)data {
+    NSDictionary *xmlDictonary = [self createXmlFromData:data];
+    if (xmlDictonary) {
+        VWResponseModel *response = [[VWResponseModel alloc] initWithXml:xmlDictonary];
+        if(!response) {
+            NSError *error = [NSError errorWithDomain:@"Can't parse XML from server"
+                                                 code:0
+                                             userInfo:nil];
+            if (self.requestStatus == kRequestWinnerPicked) {
+                [HyBidLogger debugLogFromClass:NSStringFromClass([self class]) fromMethod:NSStringFromSelector(_cmd) withMessage:@"VAPI has failure but PAPI was faster."];
+                return;
+            }
+            
+            if (self.requestStatus == kRequestBothPending) {
+                self.requestStatus = kRequestVerveResponded;
+            } else {
+                [self invokeDidFail:error];
+            }
+        } else if ([PNLiteResponseOK isEqualToString:response.status]) {
+            NSMutableArray *responseAdArray = [[NSArray array] mutableCopy];
+            
+            HyBidAd *ad = [[HyBidAd alloc] initWithVWVASTXml:xmlDictonary andWithAdSize:self.adSize];
```

Comment:
Here you actually don't need to init the HyBid ad with the XML. You can just the get String from NSData. That saves you having to parse the XML back to string inside the function.

Reply:
Made the changes. You can check my latest commit.